static/js/containers/ProfilePage_test.js

```diff
@@ -70,6 +83,182 @@ describe("ProfilePage", function() {
+          out.push(entry);
+        }
+      });
+      return out;
+    };
+
+    let fullEducation = () => {
+      let clone = _.cloneDeep(USER_PROFILE_RESPONSE);
+      [ASSOCIATE, MASTERS, DOCTORATE].forEach(level => clone.education.push({
+        "id": 2,
+        "degree_name": level,
+        "graduation_date": "1975-12-01",
+        "field_of_study": "Philosophy",
+        "school_name": "Harvard",
+        "school_city": "Cambridge",
+        "school_state_or_territory": "US-MA",
+        "school_country": "US",
+        "online_degree": false
+      }));
+      return clone;
+    };
+
+    educationSwitchSelectors.forEach( ({label, value, selector}) => {
+      it(`should confirm and let you cancel when toggling the ${label} switch on education`, () => {
+        return renderComponent(pageUrlStubs[1]).then(([, div]) => {
+          helper.store.dispatch(receiveGetUserProfileSuccess(SETTINGS.username, fullEducation()));
+
+          let toggle = div.querySelector(selector);
+          return listenForActions([
+            SET_EDUCATION_DEGREE_LEVEL,
+            SET_SHOW_EDUCATION_DELETE_ALL_DIALOG,
+          ], () => {
+            TestUtils.Simulate.change(toggle);
+            let dialog = openDialog();
+            let cancelButton = dialog.querySelector('.cancel-button');
+            TestUtils.Simulate.click(cancelButton);
+            let state = helper.store.getState();
+            let profile = state.profiles.jane.profile;
+            let entries = educationEntries(value, profile);
+            assert.equal(_.isEmpty(entries), false);
+          });
+        });
+      });
+    });
+
+    educationSwitchSelectors.forEach( ({label, value, selector}) => {
+      it(`should confirm and let you delete when toggling the ${label} switch on education`, () => {
+        return renderComponent(pageUrlStubs[1]).then(([, div]) => {
+          helper.store.dispatch(receiveGetUserProfileSuccess(SETTINGS.username, fullEducation()));
+
+          let updateProfile = fullEducation();
+          updateProfile.education = updateProfile.education.filter(entry => entry.degree_name !== value);
+          updateProfile.username = SETTINGS.username;
+
+          patchUserProfileStub.throws("Invalid arguments");
+          patchUserProfileStub.withArgs(SETTINGS.username, updateProfile).returns(
+            Promise.resolve(updateProfile)
+          );
+
+
+          let toggle = div.querySelector(selector);
+          return listenForActions([
+            SET_EDUCATION_DEGREE_LEVEL,
+            SET_SHOW_EDUCATION_DELETE_ALL_DIALOG,
+            SET_EDUCATION_DEGREE_INCLUSIONS,
+            START_PROFILE_EDIT,
+            UPDATE_PROFILE_VALIDATION,
+            REQUEST_PATCH_USER_PROFILE,
+          ], () => {
+            TestUtils.Simulate.change(toggle);
+            let dialog = openDialog();
+            let deleteButton = dialog.querySelector('.delete-button');
+            TestUtils.Simulate.click(deleteButton);
```

Comment:
Also I just realized that when the last expected action is received the promise is resolved, which means that any errors which occur afterwards, or new actions that are dispatched before the callback completes, aren't detected. I'll look into fixing that in redux-asserts